algorithmic_efficiency/spec.py

```diff
@@ -97,41 +119,54 @@ def param_shapes(self):
   def model_params_types(self) -> ParameterType:
```

Comment:
I might be out of the loop here, but isn't `model_params_types` meant to be used similar to `param_shapes` just that it provides the type instead of the shapes? Shouldn't it then also be a `@property`?

Reply:
Yep, missed that. Fixed.